### ISSUE TITLE
Extract normal vectors at boundary points

### ISSUE THREAD

**Issue by nguyenkhoa0209** [nschloe/pygmsh#459]
Hello,

I would like to extract normal vectors with pygmsh. 
I create my geometry in a similar way in your example code:

    with pygmsh.geo.Geometry() as geom:

    geom.add_polygon(
        [
            [0.0, 0.0],
            [1.0, -0.2],
            [1.1, 1.2],
            [0.1, 0.7],
        ],
        mesh_size=0.1,
    )
    mesh = geom.generate_mesh()

and I consider the points that define the polygon as the boundary points.
Is it possible to extract the normal vectors at these points? 

Thank you for your help.

**Comment by nschloe** [comment]
Are you looking for the boundary normals of the geometry or the resulting mesh? In any case, normals don't make sense at the polygon _points_.

**Comment by nguyenkhoa0209** [comment]
Thank you for your reply. I'm sorry for being not clear enough previously. I would like to calculate the normal vectors on the edges of the polygon, i.e in your example code, the normal on four sides of the polygon.

**Comment by nschloe** [comment]
Then probably `gmsh.getNormal` is for you. If you want this is pygmsh, feel free to PR.

**Comment by nguyenkhoa0209** [comment]
Thanks again for your reply. Sorry for this one more question.
Is it possible to do gmsh.model.getNormal() with a geometry defined in pygmsh. 
Indeed, I have just tried with gmsh, and to create the same geometry as yours, I use the following code:

    gmsh.initialize(sys.argv)
    gmsh.model.add("normals")
    gmsh.model.geo.addPoint(0.0, 0.0, 0)
    gmsh.model.geo.addPoint(1.0, -0.2, 0)
    gmsh.model.geo.addPoint(1.1, 1.2, 0)
    gmsh.model.geo.addPoint(0.1, 0.7, 0)
    gmsh.model.mesh.generate(2)

And then I consider the gmsh's page to get the normal: https://github.com/jeromerobert/gmsh/blob/master/demos/api/normals.py

But when I tried 

    ent = gmsh.model.getEntities(2)

the ent return [] and I could not do anything next.
